clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -4614,12 +4615,16 @@ bool CompilerInvocation::ParseLangArgs(LangOptions &Opts, ArgList &Args,
           Diags.Report(diag::err_drv_hlsl_bad_shader_unsupported)
               << VulkanEnv << T.getOSName() << T.str();
         }
-        if (Args.getLastArg(OPT_fnative_half_type)) {
+        if (Args.getLastArg(OPT_fnative_half_type) ||
+            Args.getLastArg(OPT_fnative_int16_type)) {
```

Comment:
We should probably add the `-fnative-int16-type` to `clang/test/Options/enable_16bit_types_validation_spirv.hlsl` for completeness here (I don't think we need to necessarily test every possible combination of flags for different error messages though)